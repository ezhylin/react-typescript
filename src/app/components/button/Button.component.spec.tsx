import React from 'react';
import { Button, Props } from './Button.component';
import { render } from '@testing-library/react';

describe('Button component', () => {
  const getButton = ({ children, ...extraProps }: Props) => <Button {...extraProps}>{children}</Button>;

  it('should renders', () => {
    const { asFragment } = render(getButton({ children: 'Add' }));

    expect(asFragment()).toMatchSnapshot();
  });

  it('should ', () => {});
});
