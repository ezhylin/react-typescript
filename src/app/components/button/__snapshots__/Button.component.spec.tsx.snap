// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should renders 1`] = `
<button
  class="Button"
  type="button"
>
  Add
</button>
`;

exports[`Button component should spread additional class names 1`] = `
<button
  class="Button className1 className2"
  type="button"
>
  Add
</button>
`;
